dcm4che-core/src/main/java/org/dcm4che3/io/DicomInputStream.java

```diff
@@ -708,8 +715,13 @@ public void readAttributes(Attributes attrs, int len, Predicate<DicomInputStream
                     bigEndian = prevBigEndian;
                     explicitVR = prevExplicitVR;
                 }
-            } else
-                skipAttribute(UNEXPECTED_ATTRIBUTE, "readAttributes()");
+            } else if (tag == Tag.SequenceDelimitationItem) {
+                if (length != 0) {
```

Comment:
I need to see some test data to understand whether this new `if` is an actually valid case.
Also, if this problem is also present on master, where is the PR for master?
A github issue explaining the issue would also be good.

Reply:
I've removed this from the pull request. We determined that the test that is failing likely has more problems than this. It is introducing the SequenceDeliminationItem when creates the recreates the header file while storing.